pkg/bwagreement/server.go

```diff
@@ -47,7 +48,24 @@ type Agreement struct {
 
 // NewDBManager creates a new instance of a DatabaseManager
 func NewDBManager(driver, source string, logger *zap.Logger) (*DBManager, error) {
-	db, err := dbx.Open(driver, source)
+	var db *dbx.DB
+	var err error
+	// Attempt to connect to database a few times, backing off more each time
+	// there's a failure
+	for i := 1; i < 60; i = i * 2 {
+		db, err = dbx.Open(driver, source)
+		if err != nil {
+			dur := time.Second * time.Duration(i)
+			logger.Warn("Error connecting to bwagreement database.",
+				zap.Error(err),
+				zap.Duration("backoff", dur),
+			)
+			time.Sleep(dur)
```

Comment:
I think we want to monitor context cancellation as well, otherwise we'll end up waiting for 2min before being able to cancel.

Reply:
During startup? I don't see a context being used at all a few calls up the call tree. Is there one somewhere I should go chase down?

Reply:
Ah, wait I know how to do it more cleanly. Let's move creating of the Database into Config.Run. Then server takes database as a parameter (rather than the string param). And finally add the reconnect inside Config.Run, rather than making it part of `NewDBManager`. There you also have access to ctx.

Reply:
ok, I'll see if I can do that.